reposerver/repository/repository.go

```diff
@@ -306,6 +307,14 @@ func (s *Service) runRepoOperation(
 		defer settings.sem.Release(1)
 	}
 
+	// do not generate manifests if Path and Chart fields are not set for a source in Multiple Sources
+	if hasMultipleSources && source.Path == "" && source.Chart == "" {
+		log.WithFields(map[string]interface{}{
+			"source": source,
+		}).Warnf("not generating manifests as path and chart fields are empty")
+		return nil
+	}
```

Comment:
If "path" and "chart" are missing I think we should 

1. if `ref` is set, assume the user intends to use the repo _only_ for values files, and silently skip manifest generation (no need to log a warning)
2. if `ref` is _not_ set, throw an error - the user likely forgot to set either `path` or `chart`

Reply:
We should probably also update `NormalizeSource` in util/argo/argo.go to enforce these (and any other) new `sources` field rules.